pkg/compliance/io.go

```diff
@@ -23,16 +25,18 @@ type Mgr interface {
 	GenerateComplianceReport(ctx context.Context, spec v1alpha1.ReportSpec) error
 }
 
-func NewMgr(client client.Client, log logr.Logger) Mgr {
+func NewMgr(client client.Client, log logr.Logger, config etc.Config) Mgr {
```

Comment:
Do you need to pass the whole operator config to the compliance manager? Maybe you can use [functional options](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) pattern here and just pass the limit config?
The compliance mgr should know nothing about the Operator and it's config IMO